src/java/org/apache/cassandra/service/accord/interop/AccordInteropExecution.java

```diff
@@ -277,19 +276,81 @@ private AsyncChain<Data> readChains()
                     try (PartitionIterator iterator = StorageProxy.readRegular(group, consistencyLevel, this, requestTime))
                     {
                         if (iterator.hasNext())
+                        {
+                            try (RowIterator partition = iterator.next())
+                            {
+                                TxnDataKeyValue value = new TxnDataKeyValue(partition);
+                                if (value.hasRows() || command.selectsFullPartition())
+                                    result.put(fragment.txnDataName(), value);
+                            }
+                        }
+                    }
+                    return result;
+                }));
+            });
+        });
+        return results;
+    }
+
+    private List<AsyncChain<Data>> rangeReadChains(int nowInSeconds, Dispatcher.RequestTime requestTime)
+    {
+        TxnKeyRead read = (TxnKeyRead) txn.read();
+        Seekables<?, ?> keys = txn.read().keys();
+        List<AsyncChain<Data>> results = new ArrayList<>();
+        keys.forEach(key -> {
+            read.forEachWithKey((PartitionKey) key, fragment -> {
```

Comment:
I'm struggling to see how this can work - surely this shouldn't be a partition key here?

Reply:
That is, shouldn't `txn.read()` be a `TxnRangeRead`?

Reply:
There is no live migration in this so yeah it's not reachable (and doesn't work) in `mixed` and `full` because the tests only cover the fully migrated case which will never use `AccordInteropExecution`.

I'll remove the interop execution changes and put them in the live migration patch instead.

Reply:
Sounds good